### PR TITLE
Enable playback for LiveTV on tvOS

### PR DESCRIPTION
**Pull request by jhays**
Adds support to LiveTV to utilize the new VideoPlayer. 
Playback of LiveTV channels from both the Programs view and Channels view is now working. 

Minor adjustment to VLCPlayerViewController to remove usage of a force-unwrap on runTimeTicks, which was leading to a crash when attempting liveTV playback. If value is not found, 0 will be used.

### CODE REVIEW

File: Swiftfin tvOS/Views/LiveTVChannelsView.swift [hunk 1]
```diff
@@ -55,7 +55,9 @@ struct LiveTVChannelsView: View {
 		let channel = item.channel
 		if channel.type != "Folder" {
 			Button {
-				self.router.route(to: \.videoPlayer, channel)
+                self.viewModel.fetchVideoPlayerViewModel(item: channel) { playerViewModel in
+                    self.router.route(to: \.videoPlayer, playerViewModel)
```
Comment (LePips) [inline Swiftfin tvOS/Views/LiveTVChannelsView.swift:59]: For now, literally copy the current video player and rename it to LiveTVPlayerViewController, copy the overlay and rename it to LiveTVOverlay and have the new view controller use it, make its corresponding SwiftUI LiveTVPlayer view, and fix the coordinators to use that view instead.

It's the beginning of work that we need.

**Review by LePips — CHANGES_REQUESTED** [review]

**Comment by LePips** [comment]
Also merge after https://github.com/jellyfin/Swiftfin/pull/320

File: Swiftfin tvOS/Views/LiveTVChannelsView.swift (hunk 1 above)
Comment (LePips) [reply Swiftfin tvOS/Views/LiveTVChannelsView.swift:59]: Sorry, also add TODO comments at the top as well as descriptions saying what needs to happen next with these views.

Also add a TODO comment above VideoPlayerViewController and createVideoPlayerViewController stating that a new class must be created for LiveTV objects and then consumed by the other LiveTV views.

**Comment by LePips** [comment]
Just something cool to note, this fixes a crash that I just received with the removal of the forced unwraps from `runtimeTicks`. Guess it was a false assumption, but media _should_ have a run time 🤷‍♂️

File: Swiftfin tvOS/Views/LiveTVChannelsView.swift (hunk 1 above)
Comment (jhays) [reply Swiftfin tvOS/Views/LiveTVChannelsView.swift]: I've duplicated the player and overlay for use with LiveTV. 

Tested playback on a real AppleTV- while playback is successful, I'm getting no audio. So I'll have to investigate further on that.

Comment (jhays) [reply Swiftfin tvOS/Views/LiveTVChannelsView.swift]: I have to chase further to figure out why, but appears the `selectedAudioStreamIndex` from `VideoPlayerViewModel` is getting set to -1 .

Comment (LePips) [reply Swiftfin tvOS/Views/LiveTVChannelsView.swift]: Okay, that's expected as it doesn't have a default audio stream. At this point, you may actually have to copy over `createVideoPlayerViewModel()` and make a `createLiveTVViewModel()` method to accommodate for this and setting the correct values.

Eventually a `LiveTVViewModel` will be created and used there.

Comment (jhays) [reply Swiftfin tvOS/Views/LiveTVChannelsView.swift]: I've duplicated the create viewModel for now. 
But I'm not sure what to adjust to fix audio playback within that method. There is a stereo aac audio track there, but for some reason I don't get any audio playback. 
The same item plays fine with audio in Jellyfin web interface in chrome. Playback from there or the appleTV shows up in the dashboard as a remux. 
Any ideas on what to tweak to try to fix audio playback? Thanks.

**Comment by LePips** [comment]
I think once the conflicts are resolved then the CI will get kicked in and then I can approve. For some reason, I _still_ cannot play live tv even on the web given whatever m3u playlist. However if it's working for you and will work for other, I'm more than happy to approve.

**Comment by jhays** [comment]
@LePips , my latest commit resolves merge conflicts, and although this branch does successfully play live TV for me, I still can't seem to get audio playback to work. 
Do you have any suggestions on where I can investigate further to determine why the audio track is not playing? 
Thanks.

File: Shared/Extensions/JellyfinAPIExtensions/BaseItemDto+VideoPlayerViewModel.swift [hunk 2]
```diff
@@ -166,4 +166,159 @@ extension BaseItemDto {
...
+					                                                subtitle: subtitle,
+					                                                directStreamURL: directStreamURL,
+					                                                transcodedStreamURL: transcodedStreamURL?.url,
+					                                                hlsStreamURL: hlsStreamURL,
+					                                                streamType: streamType,
+					                                                response: response,
+					                                                audioStreams: audioStreams,
+					                                                subtitleStreams: subtitleStreams,
+					                                                chapters: modifiedSelfItem.chapters ?? [],
+					                                                selectedAudioStreamIndex: defaultAudioStream?.index ?? -1,
```
Comment (LePips) [inline Shared/Extensions/JellyfinAPIExtensions/BaseItemDto+VideoPlayerViewModel.swift:305]: This is where the audio stream index gets set and then is set in the media player. I assume that the channel stream doesn't have a default audio stream.

File: Swiftfin tvOS/Views/VideoPlayer/LiveTVPlayerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,900 @@
...
+		}
+	}
+}
+
+// MARK: PlayerOverlayDelegate
+
+extension LiveTVPlayerViewController: PlayerOverlayDelegate {
+
+	func didSelectAudioStream(index: Int) {
+		vlcMediaPlayer.currentAudioTrackIndex = Int32(index)
```
Comment (LePips) [inline Swiftfin tvOS/Views/VideoPlayer/LiveTVPlayerViewController.swift:762]: (2) Through a variety of mechanisms, this is where the audio track gets set. From (1), the track index was -1 which is the index for "no audio track" in VLCKit. You might just have to remove this line.

**Comment by LePips** [comment]
@jhays see comments (1) and (2).

File: Shared/Extensions/JellyfinAPIExtensions/BaseItemDto+VideoPlayerViewModel.swift (hunk 2 above)
Comment (jhays) [reply Shared/Extensions/JellyfinAPIExtensions/BaseItemDto+VideoPlayerViewModel.swift:305]: If I put a breakpoint on 294 and take a look at this at the moment I select a liveTV channel item for playback, I am actually finding that the defaultAudioStream is set, to a stereo aac audio track. Also this track appears to have an index value of `-1`, which is how it matches up on line 202 where the default stream is selected from the audio streams array. In this case, that array also only has this same one audio stream object. 

Screen Shot 2022-01-24 at 6 14 38 PM

File: Swiftfin tvOS/Views/VideoPlayer/LiveTVPlayerViewController.swift (hunk 3 above)
Comment (jhays) [reply Swiftfin tvOS/Views/VideoPlayer/LiveTVPlayerViewController.swift]: Bingo! Removing this made liveTV audio work for me. Thanks for the help.

File: Shared/Extensions/JellyfinAPIExtensions/BaseItemDto+VideoPlayerViewModel.swift (hunk 2 above)
Comment (LePips) [reply Shared/Extensions/JellyfinAPIExtensions/BaseItemDto+VideoPlayerViewModel.swift:305]: Interesting, as I don't know how live tv objects are stored in JF in respect to a `BaseItemDto` object. Over time, you'll have to do more investigation and tailor the video players to the data from these objects.

**Review by LePips — APPROVED** [review]

**Pull request merged**